### DISCUSSION TITLE
Fehlermeldung Cannot calculate energy since battery capacity is missing

### DISCUSSION THREAD

**Discussion by Lausbua** [camueller/SmartApplianceEnabler#599] [Installation und Betrieb]
Hallo Zusammen,

ich erhalte im Eventlog zahlreich die Fehlermeldung:
Cannot calculate energy since battery capacity is missing

``
2024-04-20 07:23:31,267 WARN [MQTT Call: F-0xxxxxx0-000000000010-00-OptionalEnergySocRequest-37] d.a.s.s.OptionalEnergySocRequest [SocRequest.java:169] F-0xxxxxx0-000000000010-00: Cannot calculate energy since battery capacity is missing
2024-04-20 07:23:31,055 WARN [MQTT Call: F-0xxxxxx0-000000000010-00-SocRequest-244] d.a.s.s.SocRequest [SocRequest.java:169] F-0xxxxxx0-000000000010-00: Cannot calculate energy since battery capacity is missing
``

Immer jede Stunde und ziemlich zahlreich.

Der Ladevorgang und Ermittlung des SOC funktioniert einwandfrei.

Wo könnte der Fehler liegen?

Gruß,
Daniel

**Comment by camueller** [comment]
Am Log-Level `WARN` kannst Du erkennen, dass es sich nicht um einen Fehler, sondern "nur" um eine Warnung handelt. Diese sollte aber nur passieren, wenn das Fahrzeug gerade mit der Wallbox verbunden wurde.

Du hast nicht geschrieben, welche Version des SAE Du verwendest. Wenn der Fehler bei Dir wirklich stündlich auftritt, dann teste doch mal diese Test-Version, welche ich akatuell auch selbst im Einsatz habe und die vermutlich weitgehend der nächsten Release-Version entspricht: https://drive.google.com/file/d/12Gs3bmKZSQbHS8kyCrySz07sIdpOxTRr/view?usp=sharing

**Reply by Lausbua** [reply]
Hallo Axel,

ich nutze SAE 2.3.1, habe soeben deine Testversion eingespielt. Ich berichte dann.

Gruß,
Daniel

**Reply by Lausbua** [reply]
Habe einen Log mit DEBUG erstellt. Man erkannt das ab 06:30 Uhr jede Stunde die Warnmeldung "Cannot calculate energy since battery capacity is missing" zahlreich im Log erscheint. Es war kein Fahrzeug angesteckt.

Hängt das mit dem Zeitplan zusammen? Dieser startet ab 06:00 Uhr.

[rolling-2024-04-23.zip](https://github.com/camueller/SmartApplianceEnabler/files/15077396/rolling-2024-04-23.zip

![image](https://github.com/camueller/SmartApplianceEnabler/assets/3309871/2c2dd746-ce70-404e-b148-1963d5d2d512)

**Comment by camueller** [comment]
Die Meldungen bei Dir stammen von "toten" Requests. Diese Fehler habe ich gestern bei meiner Wallbox auch entdeckt und gefixt:

Probier mal diese Test-Version:
https://drive.google.com/file/d/1ozslAq3_wNH-1sMQ0F0yFYiYycg8EY-C/view?usp=sharing

**Reply by Lausbua** [reply]
Vielen Dank für deine neue Test-Version. Habe diese gerade eingespielt.

**Reply by Lausbua** [reply]
Hallo,

die zahlreichen Warnmeldungen (tote Requests) sind mit dieser Testversion verschwunden.

Sobald allerdings der Zeitplan aktiv wird (Start-/Endzeit) und es ist kein Fahrzeug angesteckt taucht die Warnmeldung stündlich im Log noch auf.

Habe von heute das Logfile nur im Loglevel INFO.
Falls du den TRACE benötigst kann ich die Datei die nächsten Tage nachliefern.
Morgen muss ich laden, und bei angestecktem Fahrzeug erscheint die Warnmeldung im Log nicht.

[log.txt](https://github.com/camueller/SmartApplianceEnabler/files/15119163/log.txt)

Gruß,
Daniel

**Reply by camueller** [reply]
Die Log-Zeile stammt vom OptionalEnergySocRequest, die es zum einen gar nicht geben dürfte bei nicht angestecktem Fahrzeug. Zum anderen werden sie bei Dir jede Stunde neu erzeugt - vermutlich wegen des Zeitplanes.
Insofern kann ich ohne Log-File mit Level TRACE nichts machen.

**Reply by Lausbua** [reply]
Ok, habe ein Log-File im Level TRACE erstellt. Um 14:38 Uhr habe ich abgesteckt da SOC 90% erreicht war und um 14:56:18,451 kommt ein OptionalSocRequest (trotz abgestecktem Fahrzeug) mit der anschließenden Warnmeldung.
[logfile.txt](https://github.com/camueller/SmartApplianceEnabler/files/15138317/logfile.txt)

**Reply by camueller** [reply]
Ich denke, dass ich den Fehler gefunden habe. Hier ist eine neue Test-Version:
https://drive.google.com/file/d/1Y1xpmAOFBsPUBBPjfTbpFLIf-4HcYEkP/view?usp=sharing

**Reply by Lausbua** [reply]
Habe zum Test soeben die Version eingespielt. Danke dir.

**Reply by Lausbua** [reply]
Die Warnmeldung taucht nicht mehr auf sowie keine OptionalEnergySocRequest Meldungen mehr. Der Fehler ist behoben.
